### PR TITLE
Merge the Env Config to Config for use custom env vars outside of the…

### LINKED ISSUE #227: Database config does not work for environments
#### Issue Description
I have a default config in config/database.js and a different config in config/env/testing.js. When I set the environment to testing the database configuration from testing.js is not used. When I remove the stores and models objects from database.js trails just does not detect any database configuration.

database.js:
```js
module.exports = {

  /**
   * Define the database stores. A store is typically a single database.
   *
   * Set production connection info in config/env/production.js
   */
  stores: {

    mongodbstore: {
      // should be 'create' or 'drop'
      migrate: 'create',
      uri: 'mongodb://localhost:27017/jungo',
      options: {}
    }
  },

  models: {
    defaultStore: 'mongodbstore',
    migrate: 'create'
  }
}
````

testing.js:
```js
'use strict'

module.exports = {

  trailpack: {
    disabled: [
      'repl'
    ]
  },
  database: {
    stores: {
      mongodbstore: {
        migrate: 'create',
        uri: `mongodb://${process.env.MONGO_HOST}:27017/jungo`,
        options: {}
      }
    },
    models: {
      defaultStore: 'mongodbstore',
      migrate: 'create'
    }
  }
}

```

#### Environment
- node version: 6.4.0
- version: 1.0.0-beta-1
- operating system: OSX 10.10.5

### LINKED ISSUE #228: Trails don't apply env configs for database
#### Issue Description
In all environments trails uses default configuration for `app.config.database.stores`

#### Environment
- node version: 5.12/6.3/6.4
- trails version: 1.0.4
- operating system: OSX

### PR DESCRIPTION
**Pull request by catrielmuller**
#### Description
If you try to add a custom variable in yours config file and override in some env, this not working. 
Its problable this commit solve a lot issues related to env config files.
Tell me whats you think about this.
Cheers

#### Issues
- resolves #228 
- resolves #227

### CODE REVIEW

File: lib/trails.js [hunk 1]
```diff
@@ -27,6 +53,7 @@ module.exports = {
       log: { }
     }
     const envConfig = (config.env && config.env[process.env.NODE_ENV]) || { }
+    config = this.deepMerge((config || { }), (envConfig || { }))
```
Comment (jaumard) [inline lib/trails.js]: If you do this all line bellow are useless normally because they merge env for main and log witch is already done by this line

**Comment by catrielmuller** [comment]
Sorry, Old habits die hard. New commit whit fix for es6
Referenced to #230

**Comment by jaumard** [comment]
In addition to this PR the `trailpack` repo need a PR too to remove the merge env that is there :) see https://github.com/trailsjs/trails/issues/231

**Comment by jaumard** [comment]
Also if you can add some tests to test/verify merge env is correct :)

**Comment by catrielmuller** [comment]
Sure @jaumard, let me refactor the code for simplify the process and add utest for this.

**Comment by catrielmuller** [comment]
@jaumard Tests and refactor added.

**Comment by jaumard** [comment]
Sounds good to me ! Can you also make the PR on `trailpack` module to remove the current merge for env config ? 
To relaunch travis build (because of travis env test fails) just close the PR and open it again. That will relaunch travis build

**Comment by catrielmuller** [comment]
@jaumard 
Close and Open for re run the travis build.
Remove the env config merge per trailpack in https://github.com/trailsjs/trailpack/pull/38

**Comment by scott-wyatt** [comment]
This is great! @tjwebb @konstantinzolotarev anyone want to merge and bump on NPM?

**Comment by tjwebb** [comment]
This is fine since it fixes an issue. I think for 2.0, I'll just include lodash.merge and some other util functions so that we don't need to maintain these custom helpers.

**Pull request merged**